### ISSUE TITLE
Swedish Glossary

### ISSUE THREAD

**Issue by rloqvist** [reactjs/sv.reactjs.org#6]
Adding a list of commonly used translations

| Term | Translation |
| :---: | :---:|
| Browser | Webbläsare |

Content that does not need to be translated:

| No translation required |
| :-----: |

**Comment by rloqvist** [comment]
Feel free to offer suggestions on all these translations, I would really like to have a discussion on what feels best to use, what is idiomatic, and what needs to be translated word for word..

**Comment by michaellopez** [comment]
@rloqvist As I am translating the Tutorial page, I have so far come across these words that I am unsure how to handle:

- **tutorial** - I have used the Swedish word "guide"
- **community** - I have rephrased the line to not need that word, but that might not work some times
- **setup** - I have used the Swedish word "uppsättning", but it doesn't feel 100 % accurate
- **Time Travel** - I have left as is
- **array** - I have left as is. Thought about the Swedish word "listor", thoughts?
- **editor** - as in text editor/IDE. I have used the Swedish word "textredigerare"
- **children** - as in describing the nesting of components. I have used the Swedish word "barn", but it doesn't feel right when I read it out loud.

**Comment by michaellopez** [comment]
More words from my translations of Tutorial:

- **key** - as in React key prop. I have left as is.
- **keys** - as in plural form of React key prop. I have left as is. Did try "nycklar" but that did not match well with leaving **key** untranslated.
- **top-level component** - I have used "komponent på högsta nivå"
- **destroy** - as in "destroy a component". I have used "riva ner"
- **property** - as in unabbreviated "prop". I have used "prop" because using "egenskap" or similar loses the connection with "prop".

I also noted while translating "Tutorial" that the text in English is written in such a way that quite a few sentences want to start with "Att" in Swedish. I've learnt that it is discouraged, but I can't find any sources on that fact online, any tips?

**Comment by machalvan** [comment]
@michaellopez I agree with your suggestions. Regarding starting a sentence with "Att", I haven't heard that it would be discouraged. But sometimes "För att" fits the translation so in that case that can be used. Otherwise I don't see any problems with beginning with "Att".

**Comment by michaellopez** [comment]
One more word: **issue** - I left it as is because I don't have any good translation for it.

**Comment by NaNdreas** [comment]
This is super useful. Just noted, I think there's a typo for "render" @rloqvist 
"rendrera" -> "rendera". 

Suggestion: I imagine that "highlighted" can sometimes be translated to "markerad" too, depending on the meaning in the context. (for highlighted text I would use "markerad" for example)

**Comment by michaellopez** [comment]
@NaNdreas I've heard both "rendrera" and "rendera". I'm a fan of "rendrera" when spoken and "rendera" when written 😅 I don't mind either way, but we need to be consistent throughout.

**Comment by NaNdreas** [comment]
So after my first page translated I also have some words:

- **Life cycle (methods)** - I translated to "livscykel" and "livscykelmetoder" as it sounds pretty similar.  
- **Try it in CodePen** - All CodePen links, I translated to "Prova i CodePen". 
- **(the) Output** - I left in English, "output" and "outputen". 
- **re-render** - "Rendera om"
- **UI** - Användargränssnitt
- **(The) state** - "staten". I read it is the "government" word, "svenska staten" etc, but I think it could work. 
- **Functional component** - funktionskomponent
- **class** - "klass"
- **function** - "funktion"
- **extend** - "Extendar". "It extends `React.Component`". "Den extendar `React.Component`". The alternative is "förlänga" but it's so far from the original word, and it's a key word in js. 
- **body** - "`render()` body" I translated to "`render()`-innehåll"
- **(the) constructor** - I didn't translate, "constructor(n)", but "konstruktör(en) sounds ok too since it's similar. 
- **stateless/stateful components** - I saw from a PR to use "komponenter med state" and "komponenter utan state"
- **Variable names in code** - I translated only strings and comments in code blocks. I imagine it's confusing to open the codepen links otherwise. 
- **mount(ing)/unmount(ing)** - "mount(as)/unmount(as)", I left in English. "montera/avmontera" could work, I was thinking mostly about methods "componentDidMount" and similar are easily conncted.
- **assign** - tilldela
- **merge/merged** - slå samman/sammanslagen
- **shallow** - "grund". It's about deep/shallow merge/copy in objects, looks a bit strange to me. 
- **data flows down** - "data flödar nedåt"

**Comment by michaellopez** [comment]
@NaNdreas My thoughts on your list:

- **Functional component** - I prefer "funktionell komponent", I've never header or used "funktionskomponent"
- **(the) constructor** - I use "konstruktorn" and I think it is the official swedish term for it

Other than that I agree with the rest of your suggestions.

**Comment by michaellopez** [comment]
Some more words from **Handling Events**:

- **camelCase** - "camelCase"
- **lowercase** - "gemener"
- **event handlers** - event-hanterare
- **pattern** - as in "programming patterns", used "mönster"
- **the callback** - "callbacken"
- **arrow function** - "pil-funktion"
- **loop** - "loop"

Comments, thoughts?

**Comment by NaNdreas** [comment]
Ah good catch for the "konstruktor". It doesn't exist in the official wordlist (only "konstruktör"), but indeed it is used in programming resources after some tricky searching. 

I agree "funktionell komponent" is better. 

For the "Handling Events" I also think it looks good. I left "arrow function" untranslated, but I think it's fine as "pil-funktion" too so I can change it in my PR. 

It would be great to get some of these words (maybe not all, but some) into the first post table

**Comment by michaellopez** [comment]
@lofgrenfredrik What's your opinion on `arrow function` translation?

**Comment by lofgrenfredrik** [comment]
@michaellopez I think "pil-funktion" sounds good, and I think that's what I'm using when talking about it in swedish. 👍

**Comment by rloqvist** [comment]
I am conflicted about how to approach "escape" as in escaping HTML, "fly" or "undfly" is a very direct translation, however it is not very idiomatic nor very relevant for this domain. I'm leaning more towards translations as "säkra" or "nollställ". Suggestions?

**Comment by shwilliam** [comment]
After translating the Forms page (#12) I have some doubts about the following:

- **form(s)** - "formulär" although strongly considered using "forms"
- **read-only** - "skrivskyddat"
- **logs** - as in `console.log`, used "loggar"
- **sent** - as in a form submission, used "skickas"
- **maintain a certain internal state** - "bevakar en viss intern state"

**Comment by danielpox** [comment]
@rloqvist 
> I am conflicted about how to approach "escape" as in escaping HTML, "fly" or "undfly" is a very direct translation, however it is not very idiomatic nor very relevant for this domain. I'm leaning more towards translations as "säkra" or "nollställ". Suggestions?

What about ”avväpna”, or perhaps ”escape:a”? What do you say in common speech?

**Comment by michaellopez** [comment]
I'm +1 on "escape:a", but I don't think it's common in writing. In writing in leaning more towards "säkra"

**Comment by danielpox** [comment]
”Säkra” does express the correct meaning, so that seems like the best option in my opinion.

**Comment by danielpox** [comment]
@michaellopez 
> @rloqvist As I am translating the Tutorial page, I have so far come across these words that I am unsure how to handle:
> 
> * **tutorial** - I have used the Swedish word "guide"

I’ve found translating **guide** to ”guide”, and **tutorial** to ”guidad genomgång” quite fitting.
Thoughts on this?

**Comment by danielpox** [comment]
> * **children** - as in describing the nesting of components. I have used the Swedish word "barn", but it doesn't feel right when I read it out loud.

How about translating **parent** to ”överordnad” and **child** to ”underordnad”? For example **child components** would translate to ”underordnade komponenter”, and **passing data from parent (component) to child** would render as ”skicka data från en överordnad komponent till en underordnad”.

Using ”förälder” and ”barn” doesn’t render the appropriate meaning in Swedish. Since the meaning represents members of a hierarchy it would probably be best to use concrete terms.

Any other suggestions?

**Comment by michaellopez** [comment]
@danielpox Great suggestions, I'll update my PRs ASAP.

**Comment by danielpox** [comment]
@NaNdreas

> So after my first page translated I also have some words:
> 
> * **(the) Output** - I left in English, "output" and "outputen".
> * **(The) state** - "state(n)". It looks a lot like the "government" word, "svenska staten" which bothers me when reading, but I think it could work.
> * **body** - "`render()` body" I translated to "`render()`-innehåll"
> 
> (Please feel free to challenge any/all of these with your thoughts, that's why I post them)

* **(the) input** — Similarly ”input” and ”inputen”. Perhaps ”input:en” to separate English from Swedish grammar?
* **The state** — How about ”state:et”? I find myself saying ”state:et uppdateras” rather than ”state:en”. Thoughts?
* **body** — I did some research and I found terms like ”funktionskropp” and ”kodblock”. The tricky thing is making it sound grammatically correct. ”The `render()` body” could render as ”`render()`-funktionens kodblock” or ”`render()`s funktionskropp”. Does this make sense?

**Comment by danielpox** [comment]
Thoughts that came up whilst translating the Create a New React App page:
* **single-page app** — ”single-page-app”. Any other suggestions for translating ”single-page”?
* **bundler** — ”paketerare”
* **toolchain** — ”verktygskedja”
* **the power of …** — I chose ”styrkan hos …”, since ”kraften av …” doesn’t quite convey the right meaning.
* **zero-configuration bundler** — ”nollkonfigurationspaketerare”, phew. Other ideas than this long term?
* **vast ecosystem** — ”brett ekosystem”

Do these need to be translated?
* **out-of-the-box** — Does this need to be translated? I can’t come up with a correct translation.
* **styling**
* **routing**
* **scratch**

**Comment by andriijas** [comment]
@rloqvist  hi, is it possible to get the hacktoberfest label on this repo? https://hacktoberfest.digitalocean.com/hacktoberfest-update   

tack
